crates/cli/src/bin/wasm-bindgen-test-runner/headless.rs

```diff
@@ -121,8 +121,24 @@ pub fn run(server: &SocketAddr, shell: &Shell, timeout: u64) -> Result<(), Error
 
     // Visit our local server to open up the page that runs tests, and then get
     // some handles to objects on the page which we'll be scraping output from.
-    let url =
-        std::env::var("WASM_BINDGEN_TEST_ADDRESS").unwrap_or_else(|_| format!("http://{}", server));
+    //
+    // If WASM_BINDGEN_TEST_ADDRESS is set, use it as the local server URL,
+    // trying to inherit the port from the server if it isn't specified.
+    let url = match std::env::var("WASM_BINDGEN_TEST_ADDRESS") {
+        Ok(u) => {
+            let mut url = Url::parse(&u)?;
+            if url.port().is_none() {
+                if url.set_port(Some(server.port())).is_err() {
```

Comment:
I don't believe we have to account for that error, because in that case the URL wouldn't really be usable by us, so using `unwrap()` should be fine here.

https://docs.rs/url/2.5.0/url/struct.Url.html#method.set_port